SYS_ATL/API.py

```diff
@@ -1031,6 +1031,31 @@ def par_to_seq(self, par_pattern):
 
         return Procedure(loopir, _provenance_eq_Procedure=self)
 
+    def rearrange_dim(self, alloc_pattern, dimensions):
+        if not isinstance(alloc_pattern, str):
+            raise TypeError("expected first argument to be allocation "+
+                            "pattern string")
+        if not isinstance(dimensions, list):
+            raise TypeError("expected second argument to be integer list of "+
+                            "dimensions")
+
+        stmts_len = len(self._find_stmt(alloc_pattern, default_match_no=None))
+        loopir = self._loopir_proc
+        for i in range(0, stmts_len):
+            s = self._find_stmt(alloc_pattern, body=loopir.body, default_match_no=None)[i]
+            if not isinstance(s, LoopIR.Alloc):
+                raise TypeError("pattern did not describe an alloc statement")
+            # Check that the number of dimensions matches with alloc size
+            assert type(s.type) is T.Tensor
+            if len(s.type.hi) != len(dimensions):
+                raise TypeError("dimension does not match with the dimension of "+
+                                "the alloc statement")
+
+            loopir  = Schedules.DoRearrangeDim( loopir, s, dimensions ).result()
```

Comment:
OdDlY forMaTteD **_lines_** like **this** ᴅɪsᴛʀᴀᴄᴛ readers/**reviewers** a n d **eat** up _time_.

1. The formatting of this function is internally inconsistent:
    1. Two spaces after `loopir` and before `=`?
    2. Spaces inside a function call? `f( x )` vs `f(x)`?
2. My review time is better spent discussing program logic than whitespace.
3. **Your engineering time is better spent anywhere besides fiddling with whitespace.**

On Monday, I want to re-raise the issue of mechanizing our code formatting, ref. #59

Reply:
I get you were frustrated but this comment was quite funny to me

Reply:
I _did_ try to cut my frustration with a good bit of humor 🙂